reference/api/settings.mdx

```diff
@@ -1910,3 +1911,333 @@ Reset an index's typo tolerance settings to their [default value](#typo-toleranc
+
+- `openAi`
+- `huggingFace`
+- `ollama`
+
+Additionally, use `rest` to auto-generate embeddings with any embedder offering a REST API.
+
+You may also configure a `userProvided` embedder. In this case, you must manually include vector data in your documents' `_vector` field. You must also manually generate vectors for search queries.
+
+This field is mandatory.
+
+##### `url`
+
+Meilisearch queries `url` to generate vector embeddings for queries and documents. `url` must point to a REST-compatible embedder.
+
+This field is mandatory when using `rest` embedders.
+
+This field is optional when using `ollama` embedders.
+
+This field is incompatible with `openAi`, `huggingFace`, and `userProvided` embedders.
+
+##### `apiKey`
+
+Authentication token Meilisearch should send with each request to the embedder.
+
+This field is mandatory if using a protected `rest` embedder.
+
+This field is optional for `openAI` and `ollama` embedders. If you don't specify `apiKey`, Meilisearch will attempt to read it from environment variables `OPENAI_API_KEY` and `MEILI_OLLAMA_URL`, respectively.
+
+This field is incompatible with `huggingFace` and `userProvided` embedders.
+
+##### `model`
+
+The model your embedder uses when generating vectors. These are the officially supported models Meilisearch supports:
+
+- `openAi`: `openai-text-embedding-ada-002`, `text-embedding-3-small`, and `text-embedding-3-large`
+- `huggingFace`: `BAAI/bge-base-en-v1.5`
+
+Other models, such as those provided by Ollama and REST embedders may also be compatible with Meilisearch.
+
+This field is mandatory for `openAi`, `huggingFace`, and `Ollama` embedders.
+
+This field is incompatible with `rest` and `userProvided` embedders.
+
+##### `documentTemplate`
+
+`documentTemplate` is a string containing a [Liquid template](https://shopify.github.io/liquid/basics/introduction). Meillisearch interpolates the template for each document and sends the resulting text to the embedder. The embedder then generates document vectors based on this text.
+
+You may use the following context values:
+
+- `{{doc.FIELD}}`: `doc` stands for the document itself. `FIELD` must correspond to an attribute present on all documents value will be replaced by the value of that field in the input document
+- `{{fields}}`: a list of all the `field`s appearing in any document in the index. Each `field` object in this list has two properties: `name` and `value`. If a `field` does not exist in a document, `value` is `nil`
+
+For best results, build short templates that only contain highly relevant data. If working with a long field, consider [truncating it](https://shopify.github.io/liquid/filters/truncatewords/). If you do not manually set it, `documentTemplate` will include all document fields. This may lead to suboptimal performance and relevancy.
+
+This field is optional for all embedders.
```

Comment:
We should maybe specify that while optional, it is recommended to fill it because the default template may render to long texts and contain information that is not relevant to the semantics of the document (such as the id of the document)